### ISSUE TITLE
About the model

### ISSUE THREAD

**Issue by WuXingrui** [edwardnguyen1705/robotic-grasping-cornell#1]
**Thanks for your supply! 
when tryed to download the pretrained model ,i found the link seems to be null.I can't download the model now.
And i got the model_100.ckpt after the train step, when i use it in the demo.py get the error like this:**
RuntimeError: Error(s) in loading state_dict for GraspNet:
 Missing key(s) in state_dict: "GraspNet_base.0.weight", "GraspNet_base.0.bias", "GraspNet_base.2.weight", "GraspNet_base.2.bias", "GraspNet_base.5.weight", "GraspNet_base.5.bias", "GraspNet_base.7.weight", "GraspNet_base.7.bias", "GraspNet_base.10.weight", "GraspNet_base.10.bias", "GraspNet_base.12.weight", "GraspNet_base.12.bias", "GraspNet_base.14.weight", "GraspNet_base.14.bias", "GraspNet_base.17.weight", "GraspNet_base.17.bias", "GraspNet_base.19.weight", "GraspNet_base.19.bias", "GraspNet_base.21.weight", "GraspNet_base.21.bias", "GraspNet_base.24.weight", "GraspNet_base.24.bias", "GraspNet_base.26.weight", "GraspNet_base.26.bias", "GraspNet_base.28.weight", "GraspNet_base.28.bias", "GraspNet_classifier.0.weight", "GraspNet_classifier.0.bias", "GraspNet_classifier.3.weight", "GraspNet_classifier.3.bias", "GraspNet_regressor.0.weight", "GraspNet_regressor.0.bias", "GraspNet_regressor.3.weight", "GraspNet_regressor.3.bias", "GraspNet_cls_score.weight", "GraspNet_cls_score.bias", "GraspNet_rect_pred.weight", "GraspNet_rect_pred.bias". 
 Unexpected key(s) in state_dict: "epoch", "model", "optimizer", "loss". _

**Comment by WuXingrui** [comment]
Hoping to get some help from you!
And I'm doing sone job about the robot grasp,can you share sone experience?

**Comment by edwardnguyen1705** [comment]
Hi @WuXingrui , I am sorry for the confusion. 
1. I already check the pretrained model link, you can download it now.
2. If you train by yourself and get the model name model_100.ckpt, goto the line 34 in the demo.py, please change the model name model.ckpt --> model_100.ckpt. Just before I named the model model.ckpt, later I put the epoch info. If you have any problem, please just let me know.

**Comment by SHYang1210** [comment]
Hi, hi, I changed the model name on line 34 of demo.py, but this error still occurs. i can run for the model you provide. hope to get your help.
RuntimeError: Error(s) in loading state_dict for GraspNet:
Missing key(s) in state_dict: "GraspNet_base.0.weight", "GraspNet_base.0.bias", "GraspNet_base.2.weight", "GraspNet_base.2.bias", "GraspNet_base.5.weight", "GraspNet_base.5.bias", "GraspNet_base.7.weight", "GraspNet_base.7.bias", "GraspNet_base.10.weight", "GraspNet_base.10.bias", "GraspNet_base.12.weight", "GraspNet_base.12.bias", "GraspNet_base.14.weight", "GraspNet_base.14.bias", "GraspNet_base.17.weight", "GraspNet_base.17.bias", "GraspNet_base.19.weight", "GraspNet_base.19.bias", "GraspNet_base.21.weight", "GraspNet_base.21.bias", "GraspNet_base.24.weight", "GraspNet_base.24.bias", "GraspNet_base.26.weight", "GraspNet_base.26.bias", "GraspNet_base.28.weight", "GraspNet_base.28.bias", "GraspNet_classifier.0.weight", "GraspNet_classifier.0.bias", "GraspNet_classifier.3.weight", "GraspNet_classifier.3.bias", "GraspNet_regressor.0.weight", "GraspNet_regressor.0.bias", "GraspNet_regressor.3.weight", "GraspNet_regressor.3.bias", "GraspNet_cls_score.weight", "GraspNet_cls_score.bias", "GraspNet_rect_pred.weight", "GraspNet_rect_pred.bias".
Unexpected key(s) in state_dict: "epoch", "model", "optimizer", "loss". _

**Comment by edwardnguyen1705** [comment]
Hi @SHYang1210 ,
Sorry for the confussion.
At the moment,  I have modified the `demo.py`, but my laptop does not have GPU so I cannot test.
Please follow this tutorial to understand the saving and loading mechanisms: [https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-a-general-checkpoint-for-inference-and-or-resuming-training](url) 
Especially this section, "Saving & Loading a General Checkpoint for Inference and/or Resuming Training".
Hope that help.

**Comment by mouad47** [comment]
Hi @edwardnguyen1705  thank you for this repository, it was very helpful.
but i have a little problem and i will appreciate your help.
the problem is that I'm working with keras, and it's require that the data should be in xml or csv format, so do you have any idea about that !
thank you

**Comment by edwardnguyen1705** [comment]
Hi @mouad47 ,

You need either write a customized dataloader or check the file `grasp_dataset.py` to convert the dataset to `xml` format refer to this repo [https://github.com/AndrewCarterUK/pascal-voc-writer](url)

**Comment by mouad47** [comment]
yeah i see it, what makes me more uncomfortable with it it's that the grasp dataset has 3 line in the txt file,that mean 3 annotation or three rectangle, so i have to convert all the rectangles !

**Comment by mouad47** [comment]
thank you @edwardnguyen1705 i solevd the problem, 
another problem has shaowed if you have any idea please.
the file is in the directory but i have this error: 
[Errno 2] No such file or directory: '../PFE_Project/CGD/Images/pcd0116r_preprocessed_5.jpg'

**Comment by edwardnguyen1705** [comment]
@mouad47  you should generate data by yourself

**Comment by mouad47** [comment]
thank you @edwardnguyen1705 
the project is done, i used your dataset, but i create my own xml converter, thank you you were very helpful, 
another thing that is really i don't understand, where the depth images ! should we use them as rgd format, thank by the way,

**Comment by edwardnguyen1705** [comment]
Hi @mouad47 ,
I do not use the depth images.
